domain/model/service/modelservice.go

```diff
@@ -162,25 +178,44 @@ func (s *ModelService) CreateModel(
 	ctx context.Context,
 	controllerUUID uuid.UUID,
 ) error {
-	return errors.Capture(s.CreateModelForVersion(ctx, controllerUUID, agentVersionSelector()))
+	if err := createModelForVersion(
+		ctx, s.modelID, controllerUUID, s.agentBinaryFinder, agentVersionSelector(), s.controllerSt, s.modelSt,
+	); err != nil {
+		return errors.Capture(err)
+	}
+
+	env, err := s.providerGetter(ctx)
+	if err != nil {
```

Comment:
There exists the possibility that we will get back a not implemented error here. We should handle this and do nothing.

Reply:
This! Providers are not consistent at all. They either have no method to state they don't support it OR they have the method and return not supported. We're wildly inconsistent here.

Reply:
yeah, fixed!